feature-utils/poly-look/test/react-components/chip.test.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import Chip from "../../src/react-components/filterChips/chip.jsx";
+
+const mockedHandleClick = jest.fn();
+
+const chip = <Chip id="chipId" handleClick={mockedHandleClick} active={true} />;
+
+it("Renders Chip", () => {
+  const { container, getByRole } = render(chip);
+  expect(container).toBeTruthy();
+  fireEvent.click(getByRole("button"), mockedHandleClick);
```

Comment:
Since this is testing behaviors, it's not so easy to select by class or ID at this level. You can select by role, which is related to behavior or to what's called an [ARIA role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles). The whole point of using roles is that you can change the specific element, as long as you keep the role (by just using the attribute `role=button`)

Reply:
Also, relying on specific DOM structures might not be a good idea. That could change (as it did when we were testing poly-explorer), it's brittle and it's not really testing behaviors.

Reply:
As you will see in another comment below, I changed it for `getByText` and I used the chip id to select it. Would that change be an improvement then?